src/passes/OptimizeInstructions.cpp

```diff
@@ -1634,10 +1674,8 @@ struct OptimizeInstructions
     if (auto* left = binary->left->dynCast<Binary>()) {
       if (auto* right = binary->right->dynCast<Binary>()) {
         if (left->op != right->op &&
-            ExpressionAnalyzer::equal(left->left, right->left) &&
-            ExpressionAnalyzer::equal(left->right, right->right) &&
-            !effects(left->left).hasSideEffects() &&
-            !effects(left->right).hasSideEffects()) {
+            equalAndRemovable(left->left, right->left) &&
+            equalAndRemovable(left->right, right->right)) {
```

Comment:
This part is a functional change because it now checks only the fallthrough values, right? Does this interact with GC at all?